docarray/array/array_stacked.py

```diff
@@ -162,12 +162,24 @@ def _set_array_attribute(
             setattr(self._docs, field, values)
 
     def __getitem__(self, item):  # note this should handle slices
+        if isinstance(item, slice):
+            return self._get_slice(item)
         doc = self._docs[item]
         # NOTE: this could be speed up by using a cache
         for field in self._columns.keys():
             setattr(doc, field, self._columns[field][item])
         return doc
 
+    def _get_slice(self: T, item: slice) -> T:
+        """Return a slice of the DocumentArrayStacked
+
+        :param item: the slice to apply
+        :return: a DocumentArrayStacked
+        """
+        docs = self._docs[item].stack()
```

Comment:
I think there is a problem here when you call stack you will perform the stacking on the the slice of the Nornal DocumentArray but this is useless because this slice has already been stacked into the mother one.

I believe what should be done instead is to create a constructor on DocumentArrayStacked that allow to construct a da stacked from a normal document array (the skeleton) and with the column prebuilt.

somethling lime

```python
@classmethod
def _from_columns(cls: Type[T], docs: DocumentArray, columns: Dict[..., ...])->T.
    da_stacked = DocumentArray[cls.document_type]([]).stack()
    da_stacked._columns = columns
    da_stacked._docs = docs
```
```python
def _get_slice(self: T, item: slice) -> T:
    columns = {k: col[item] for k, col in self._columns.items()}
    docs = self.__cls__._from_columns(self._docs[item], columns)
```

Reply:
So what is the problem with the current implementation?
I agree that it is "useless" (basically a no-op that just converts to the other type), but that is exactly what I want her